File: caller/caller.rb
require 'json'

def call(interpreter, filepath, *args)
  ret = ''
  cmd = ([interpreter, filepath] + args).join(' ')
  begin
    # ret = IO.popen(cmd).gets
    ret = `#{cmd}`
  rescue Exception => e
    puts "Error while executing: #{cmd}"
    puts e.message
    puts e.backtrace.inspect
    ret = ''
  ensure
    # TODO, I dont know what to do.
  end
  # print(cmd)
  ret.gsub!('\'', '"')
  ret = JSON.parse(ret)
  return ret
end

def call_py(filepath, *args)
  interpreter = 'python'
  return call(interpreter, filepath, args)
end

File: caller/caller.py
import json
import os

def call(interpreter, filepath, bejson=True, *args):
  ret = ''
  cmd = ' '.join([interpreter, filepath] + list(args))
  try:
    ret = os.popen(cmd).readlines()
    ret = ''.join(ret)
  except Exception as e:
    print('Error while executing: %s' % cmd)
    print(e)
  # try
  ret.replace('\'', '"')
  if bejson: ret = json.loads(ret)
  return ret
# call

if __name__ == '__main__':
  print(call('python', 'test.py', False))